renderer/HardwareOpenGL.cpp

```diff
@@ -166,10 +149,8 @@ ushort *opengl_packed_Upload_data = NULL;
 ushort *opengl_packed_Translate_table = NULL;
 ushort *opengl_packed_4444_translate_table = NULL;
 
-rendering_state OpenGL_state;
-static float Alpha_multiplier = 1.0f;
-
-renderer_preferred_state OpenGL_preferred_state = {0, 1, 1.5};
+extern rendering_state gpu_state;
```

Comment:
sharing state using `extern` variables everywhere makes it harder to understand. I hope we can use real dependency injection at some point.

Reply:
Agreed, I'm just following the existing pattern everywhere, and just want it to work before we do big architectural changes. As mentioned above, this will need to be better organized for other renderers to work at the same time, or even at all, but let's start slow :).